source/connect/connection-options.txt

```diff
@@ -42,426 +50,1319 @@ connection options in the string as ``<name>=<value>`` pairs. In the following e
+      See :manual:`Write Concern <reference/write-concern/>` in the
+      {+mdb-server+} manual for more information.
+
+      .. literalinclude:: /includes/fundamentals/code-examples/connection/ConnectionOptions.cs
+         :language: csharp
+         :copyable: true
+         :dedent: 8
+         :start-after: // start-settings-write-concern
+         :end-before: // end-settings-write-concern
+
+   .. tab:: MongoUrlBuilder
+      :tabid: mongo-url-builder
+
+      FSync
+      ~~~~~
+
+      The FSync component of the write concern. The default value is ``false``. To learn
+      more about the ``fsync`` command, see :manual:`fsync </reference/command/fsync/>`
+      in the {+mdb-server+} manual.
+
+      .. literalinclude:: /includes/fundamentals/code-examples/connection/ConnectionOptions.cs
+         :language: csharp
+         :copyable: true
+         :dedent: 8
+         :start-after: // start-builder-fsync
+         :end-before: // end-builder-fsync
+
+      Journal
+      ~~~~~~~
+
+      The ``j`` component of the write concern, which requests acknowledgment that the
+      MongoDB instances have written to the on-disk journal. The default value depends on
+      the value in the ``writeConcernMajorityJournalDefault`` setting. To learn more about the
+      ``j`` option, see :manual:`Write Concern </reference/write-concern/#j-option>`
+      in the {+mdb-server+} manual.
+
+      .. literalinclude:: /includes/fundamentals/code-examples/connection/ConnectionOptions.cs
+         :language: csharp
+         :copyable: true
+         :dedent: 8
+         :start-after: // start-builder-journal
+         :end-before: // end-builder-journal
+
+      W
+      ~
+
+      The ``w`` component of the write concern, which requests acknowledgment that the write
+      operation has propagated to a specified number of MongoDB instances. The default
+      value is ``"majority"`` or ``1``, depending on the number of arbiters and voting nodes.
+      To learn more about the ``w`` option, see :manual:`Write Concern </reference/write-concern/#std-label-wc-w>`
+      in the {+mdb-server+} manual.
+
+      .. literalinclude:: /includes/fundamentals/code-examples/connection/ConnectionOptions.cs
+         :language: csharp
+         :copyable: true
+         :dedent: 8
+         :start-after: // start-builder-w
+         :end-before: // end-builder-w
+
+      WTimeout
+      ~~~~~~~~
+
+      The ``wtimeout`` component of the write concern, which specifies a time limit for the
+      write concern. To learn more about the ``wtimeout`` option, see
+      :manual:`Write Concern </reference/write-concern/#wtimeout>`
+      in the {+mdb-server+} manual.
+
+      .. literalinclude:: /includes/fundamentals/code-examples/connection/ConnectionOptions.cs
+         :language: csharp
+         :copyable: true
+         :dedent: 8
+         :start-after: // start-builder-w-timeout
+         :end-before: // end-builder-w-timeout
```

Comment:
Q: does C# have CSOT yet? Some timeout options such as this one might be deprecated?

Reply:
not yet!